File: problems/tools/ListNode.py
from typing import Optional


class ListNode:
    val: int 
    next: Optional['ListNode']
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

    def __str__(self) -> str:
        return f"{self.val}->{self.next}"

    def build(arr: list[int]) -> Optional['ListNode']:
        if arr:
            head = ListNode(val=arr.pop(0))
            node = head 
            while arr:
                node.next = ListNode(val=arr.pop(0))
                node = node.next 
            return head 
        
        

File: problems/tools/ListNode.kt
package problems.tools

class ListNode(var `val`: Int = 0, var next: ListNode? = null) {
    override fun toString(): String = "$`val`->$next"

    companion object {
        fun build(values: Iterable<Int>): ListNode? {
            val iterator = values.iterator()
            if (!iterator.hasNext()) return null

            val head = ListNode(iterator.next())
            var node = head
            while (iterator.hasNext()) {
                node.next = ListNode(iterator.next())
                node = node.next!!
            }
            return head
        }
    }
}
